src/tir/transforms/lower_sparse_iter.cc

```diff
@@ -658,6 +668,7 @@ class IterTransformer : public StmtExprMutator {
           Map<String, ObjectRef> annotations;
           annotations.Set("sparse", Bool(true));
           annotations.Set("preprocess", Bool(true));
+          annotations.Set("is_binary_search_block", Bool(true));
```

Comment:
I have a small request: adding an argument `check_invalid_binary_search` (with default value `False`) to `LowerSparseIter` function, and only emit this when this argument is set to `True`, as this feature is still in its alpha stage.